index.js

```diff
@@ -112,10 +112,12 @@ module.exports = class Hyperswarm extends EventEmitter {
   }
 
   _shouldConnect () {
-    return !this.destroyed &&
-      this._connecting < this.maxParallel &&
+    const belowConnectionLimit = this._connecting < this.maxParallel &&
       this._allConnections.size < this.maxPeers &&
       this._clientConnections < this.maxClientConnections
+    return !this.destroyed &&
+      ((this._queue.length && this._queue.head().ignoreLimits) ||
```

Comment:
this seems brittle to me

Reply:
With TOP_PRIO that check should suffice to know if there are any pending peers that should ignore limits.